### DISCUSSION TITLE
HumanBodyBones를 불러오는 방법?

### DISCUSSION THREAD

**Discussion by matsu456123** [naverz/zepeto-studio-kor#1132] [5. Q&A]
`let targetCharacterAnimator = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.ZepetoAnimator;
    let targetTransform = targetCharacterAnimator.GetBoneTransform(HumanBodyBones.RightHand);`

이 방법으로 캐릭터 오른손에 뭔가를 쥐게 하고 싶은데요,

Cannot find name 'HumanBodyBones'. (at D:/Zepeto/Assets/Scripts

이렇게 오류가 뜹니다.

뭔가를 import 해야하는 것 같은데 잘 모르겠네요 스크립트 초보입니다...

아, 튜토리얼처럼 import해야할 때 자동으로 추가가 안되던데 원래 visual studio로는 안되나요?

**Comment by BumBall-Maker** [comment]
visual studio code로 작업하시는게 더 좋습니다. 오류검출도 잘 되구요
HumanBodyBones는 UnityEngine에 있습니다

import { Animator, Canvas, AnimationClip, WaitForSeconds, GameObject, Object, Random, Transform, Vector3, Quaternion, HumanBodyBones } from 'UnityEngine';

이런식으로 import하시면 됩니다.

**Reply by matsu456123** [reply]
답변 정말 감사합니다! Visual studio code를 다운받아야겠습니다. 덕분에 제대로 import 했습니다

**Comment by matsu456123** [comment]
추가질문인데요,

`System.Exception: file:///D:/Zepeto/Assets/Scripts/glassWear.js:9: TypeError: Cannot read property 'zepetoPlayer' of null`

이 오류가 계속 뜨는데 무엇을 놓친것인지 잘 모르겠습니다.

`ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
            this.zepetoCharacter = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character;
        }); `

`let targetTransform = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer.character.ZepetoAnimator;
        this.glasses.gameObject.transform.SetParent(targetTransform.GetBoneTransform(HumanBodyBones.RightHand));`

이걸로는 안되는걸까요?  glasses는 손에 쥐게 할 오브젝트 이름입니다.

**Reply by matsu456123** [reply]
이유는 모르겠는데 해결됐습니다...!
하지만 어떻게 해결된건진 정말 모르겠네요 ^^;;

**Reply by BumBall-Maker** [reply]
다행입니다! 즐거운 개발 되시길!
